### PR TITLE
Migrate to mdBook

### LINKED ISSUE #7: Usage of Vuepress' Custom Containers
Example:

image

Vuepress plugin: https://vuepress.github.io/en/plugins/container/#render

### LINKED ISSUE #19: Gray background headers
As suggested by @ISSOtm to potentially fix/improve the situation described in #18 , consider gray banded brackground heading/subheadings, as in nocash docs:

image

This requires (and thus it's blocked by) #18 consistency in the usage of headings (h1-h4)

### LINKED ISSUE #58: Empty sidebar?
Remove the sidebar if we don't plan on using it

### LINKED ISSUE #60: ToC depth
We need to revision and possibily rethink how deep is the ToC, since it provides the main way to navigate the document.

From a comment on Discord:
> Yeah, I think that is a key issue in finding info quicky. The video section has only 1 main header but contains a lot of information in subheaders.

### LINKED ISSUE #289: Easier navigation of Pan Docs
I often have multiple tabs open that each have https://gbdev.io/pandocs/ loaded. Right now, the page title is 
Screen Shot 2021-05-16 at 10 55 49 AM

Being able to dynamically update the window/tab title to include the anchored section would be great. For example, I think something akin to 
Screen Shot 2021-05-16 at 10 57 05 AM would be perfect for https://gbdev.io/pandocs/#lcd-monochrome-palettes 

Alternatively, being able to track several sections all at once and easily jump between them via hotkeys would be amazing as well. I don't have the ability to mock it up, but I'm imagining a + next to each anchor link which would add an anchor link to a little floating side bar on the left. That way, I can just quickly navigate between sections I want to remember.

Some sort of sane hotkey system to navigate between "bookmarked" sections would be a great addition to that.

### LINKED ISSUE #290: Multiple page version
Pandocs is currently being served as a one long page, following the spirit of the original document.

Some users (as #289 ) expressed the desire to be able to navigate the document split in different pages (one per chapter?). It should be trivially possiblle to add another HTML export target having one chapter / subchapter per page, while also keeping the current version.

### PR DESCRIPTION
**Pull request by avivace**
Why:
- single and multi page layout
- themes, palettes, better theming
- custom backends
- custom language highlight.js 

What's missing:
- [x] info/warning boxes renderer
- [x] Styling those boxes
- [x] Re-wiring CI
- [x] Fixing links
- [x] Custom highlight.js
- [x] Making heading levels consistent
- [x] Make info/warning boxes render on print

Fixes #289
Fixes #290
Fixes #7, since we're not using Vuepress anymore
Fixes #19 by virtue of using custom themes, though a custom "Pan Docs"/"Classic" theme could be added.
Fixes #58 due to an entirely different layout
Fixes #60

### CODE REVIEW

File: docs/.nojekyll [hunk 1]
```diff
@@ -0,0 +1 @@
+This file makes sure that Github Pages doesn't process mdBook's output.
```
Comment (avivace) [inline docs/.nojekyll:1]: This file is not needed, it's sufficient to not enable github pages with jekyll in the repository setting and github pages will just serve the selected folder / branch

File: docs/About [hunk 2]
```diff
@@ -0,0 +1 @@
+# About
```
Comment (avivace) [inline docs/About:1]: why is the output in this branch?

Comment (ISSOtm) [reply docs/About:1]: Missing `.gitignore`...

File: docs/.nojekyll (hunk 1 above)
Comment (ISSOtm) [reply docs/.nojekyll]: It's automatically generated, and I'm not seeing a way to disable it.

**Comment by ISSOtm** [comment]
@avivace How are we going to handle the custom styling of the boxes? Also, should we offer the current Pan Docs color scheme as a theme? (If yes, there are a lot of corollary questions.)

**Comment by ISSOtm** [comment]
Misc. changes to be made:

- LCDC.md, line 57 says "further below", but this is not *below* anymore.
- Video_Display.md is huge, and generates a beeg chapter without a ToC, so it should be split into multiple sub-chapters. (We can use the navbar's collapsing feature to avoid making it huge as well.)
- SGB_Functions.md is huge as well.
- CPU_Comparison_with_Z80.md compares with Z80 *and* 8080, so a slightly more generic name would be more appropriate.
- "MID-FRAME QUIRKS" box isn't marked up in the Video Display chapter.

**Comment by ISSOtm** [comment]
How should glossary entries be marked up?

**Comment by ISSOtm** [comment]
[A preview is now live](https://eldred.fr/pandocs)! Feedback is welcome.

**Comment by ISSOtm** [comment]
Note: to test CI, as well as to update the preview, I am now pushing to <https://github.com/iSSOtm/pandocs/tree/master> instead.

**Comment by ISSOtm** [comment]
I also consolidated the build and deploy CI workflows into a single one, so that the build's results can be immediately used. Deployment is skipped when not pushing, pushing to a branch that isn't `master`, or if a SSH deploy key hasn't been configured.

**Comment by ISSOtm** [comment]
Since this has a custom PDF export (via the browser), can we consider #57 fixed by this?

**Comment by ISSOtm** [comment]
Everything is done besides the glossary markup, but that still has to be added separately, so there's no point in blocking the rest of the improvements. I'm waiting on @mid-kid's discussion on #18, and any other feedback that may surface.

If no opposition is brought up, then I will push [the changes](https://github.com/iSSOtm/pandocs/) to this branch and merge them.

**Comment by avivace** [comment]
Great work @ISSOtm . This is gorgeous.

The Glossary markup is not a priority for the migration PR. It wasn't there before, it's not necessary to have it now before migrating, it could be secondary task after we migrate to the new rust-based pipeline.

**Comment by avivace** [comment]
> Since this has a custom PDF export (via the browser), can we consider #57 fixed by this?

Nope, I think we already brought that up but we could've done that before too and the PDF exported by a print from the browser does not provide a decent table of contents, etc so that issue still stands.

**Comment by ISSOtm** [comment]
> > Since this has a custom PDF export (via the browser), can we consider #57 fixed by this?
> 
> Nope, I think we already brought that up but we could've done that before too and the PDF exported by a print from the browser does not provide a decent table of contents, etc so that issue still stands.

Well, it's a hack, but since we have control over the renderer, so I can either edit `print.html` in-place, or generate an alternative version with a ToC.
Alternatively, I'll see if there is a PDF renderer for mdBook, or one to a format that can be easily converted to PDF. I've seen ePub, for example, but I don't know.

By the way, since we have a ToC for navigating to a page, and the pages themselves are small enough (except "SGB Functions", which probably needs to be split), should we consider #60 fixed?

**Comment by ISSOtm** [comment]
[mdbook-latex](https://crates.io/crates/mdbook-latex) is worth looking into; if we can't get it to work, its README links to other projects that look potentially useful.

**Comment by avivace** [comment]
> > > Since this has a custom PDF export (via the browser), can we consider #57 fixed by this?
> > 
> > 
> > Nope, I think we already brought that up but we could've done that before too and the PDF exported by a print from the browser does not provide a decent table of contents, etc so that issue still stands.
> 
> Well, it's a hack, but since we have control over the renderer, so I can either edit `print.html` in-place, or generate an alternative version with a ToC.
> Alternatively, I'll see if there is a PDF renderer for mdBook, or one to a format that can be easily converted to PDF. I've seen ePub, for example, but I don't know.
> 
> By the way, since we have a ToC for navigating to a page, and the pages themselves are small enough (except "SGB Functions", which probably needs to be split), should we consider #60 fixed?

I agree on closing #60. For the PDF export, I still want to consider merging everything into a single MD/latex file and the render an actual PDF (and HTML, for the single page export) from there.

**Comment by ISSOtm** [comment]
For that, I think the renderer I just linked could do a good job. We should look into it after merging this PR to avoid scope creep, though.

**Pull request merged**